### DISCUSSION TITLE
Unable to use systemd drop-in files with templated containers

### DISCUSSION THREAD

**Discussion by ekohl** [containers/podman#22939] [Q&A]
I'm trying to convert the [docker-compose.yaml](https://raw.githubusercontent.com/gramps-project/gramps-web-docs/main/examples/docker-compose-base/docker-compose.yml) file to quadlet files. As you can see, there are 2 containers that are mostly the same, except for 2 statements (expose port & exec).

My attempt is to create `/etc/containers/systemd/gramps@.container` as follows:
```ini
[Unit]
Description=GRAMPS %i
Requires=gramps-redis.service
After=gramps-redis.service

[Container]
Image=ghcr.io/gramps-project/grampsweb:latest

ContainerName=gramps-%i
Network=gramps.network

Volume=gramps_users:/app/users  # persist user database
Volume=gramps_index:/app/indexdir  # persist search index
Volume=gramps_thumb_cache:/app/thumbnail_cache  # persist thumbnails
Volume=gramps_cache:/app/cache  # persist export and report caches
Volume=gramps_secret:/app/secret  # persist flask secret
Volume=gramps_db:/root/.gramps/grampsdb  # persist Gramps database
Volume=gramps_media:/app/media  # persist media files
Volume=gramps_tmp:/tmp

Environment=GRAMPSWEB_TREE=Gramps Web
Environment=GRAMPSWEB_CELERY_CONFIG__broker_url=redis://gramps-redis:6379/0
Environment=GRAMPSWEB_CELERY_CONFIG__result_backend=redis://gramps-redis:6379/0
Environment=GRAMPSWEB_RATELIMIT_STORAGE_URI=redis://gramps-redis:6379/1

[Service]
Restart=always
```

Then I also created `/etc/containers/systemd/gramps@web.container.d/port.conf`:
```ini
[Container]
PublishPort=8083
```

And `/etc/containers/systemd/gramps@celery.container.d/exec.conf`:
```ini
[Container]
Exec=celery -A gramps_webapi.celery worker --loglevel=INFO
```

However, these aren't respected. My interpretation of `man podman-systemd.unit` implies this should work. Am I doing something wrong or is this not supported?

```console
# rpm -qv podman
podman-5.1.0-1.fc40.x86_64
```

**Comment by jbtrystram** [comment]
I am running into the same issue. 
The code suggest the drop-ins are loaded from the same directory as the units : https://github.com/containers/podman/blob/7ff1494c47d2535ffc80e55854a2ee02ebcbec2f/cmd/quadlet/main.go#L260

The test code suggests the same : https://github.com/containers/podman/commit/8ee26220281bc5bbb6a2b72ab0cf5a2f120b1afa

It looks like a bug, maybe the tests are broken

**Comment by Luap99** [comment]
Templates are special because by default quadlet is just a generator and cannot know what instances you have started via systemd. As such the special symlink as descriped in the docs there is required.

From https://docs.podman.io/en/latest/markdown/podman-systemd.unit.5.html#template-files

> Secondly, if you make a symlink like “foo@instance.container”, that will generate an instantiated template file. When generating this file quadlet will read drop-in files both from the instanced directory (foo@instance.container.d) and the template directory (foo@.container.d). This allows customization of individual instances.

**Reply by ekohl** [reply]
It's not obvious to me where I need to create that symlink
